UnitySDK/Assets/ML-Agents/Scripts/InferenceBrain/GeneratorImpl.cs

```diff
@@ -312,11 +312,17 @@ public class VisualObservationInputGenerator : TensorGenerator.IGenerator
         public void Generate(
             TensorProxy tensorProxy, int batchSize, IEnumerable<Agent> agents)
         {
-            var textures = agents.Select(
-                agent => agent.Info.visualObservations[m_Index]).ToList();
-
             TensorUtils.ResizeTensor(tensorProxy, batchSize, m_Allocator);
-            Utilities.TextureToTensorProxy(textures, tensorProxy, m_GrayScale);
+            var agentIndex = 0;
+            foreach (var agent in agents)
+            {
+                // TODO direct access to sensors list here - should we do it differently?
```

Comment:
It could be beneficial to group like sensors by type, instead of relying on an index into a list on each agent.  So you can just iterate through all of the CameraSensors of all agents instead of `agent.m_Sensors[index]`.  Though I'd have to look at the code more to see if that's a feasible option.

Reply:
Ideally, it does not mater if the sensor is a camera or not. We should ustt make sure there is an ordered mapping between python and inference. We could all each sensor input `sensor_<n>` where n is the sensor index. Another way to go would be to name each sensor.

Reply:
I think my suggestion was more around how we organize our data.  Out of the scope of this PR.

Reply:
@vincentpierre Do you think this is a blocker for this PR? I agree we need a more concrete mapping, but not sure if it can wait until I make a second pass on this.